src/modules/auth/components/change-password-form.vue

```diff
@@ -44,17 +45,20 @@ defineProps({
   isLoading: { type: Boolean, default: false },
 });
 
-const changePasswordData = ref(new ChangePasswordData());
+const { handleSubmit, errors } = useForm({
+  validationSchema,
+});
 
-function handleConfirm(): void {
-  emit('confirm', changePasswordData.value);
-}
+const { value: email } = useField<string>('email');
+const { value: newPassword } = useField<string>('newPassword');
+const { value: confirmPassword } =
+  useField<string>('confirmPassword');
```

Comment:
Nie korzystałem z tej libki, tylko spekuluję, czy required jest ten przekazany typ? W React są podobne hooki i jak ustawiamy jakieś default value to TS sam się domyśla jakiego jest typu, wiadomo to nie to samo ale tak sobie głośno myslę  - przyokazji się dowiem :D

Reply:
No właśnie nie jest required, ale bez tego zwraca mi unknown'a i świeci się template jak przekazuje v-model. Mogłem użyć tego generyka, albo mogłem ew. robić cast v-model="email as string". Generyk wydaje mi się lepszym rozwiązaniem:D